core/domain/wipeout_service.py

```diff
@@ -0,0 +1,119 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Services for user data."""
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from constants import constants
+from core.domain import collection_services
+from core.domain import exp_fetchers
+from core.domain import exp_services
+from core.domain import skill_services
+from core.domain import topic_services
+from core.domain import user_services
+from core.platform import models
+
+current_user_services = models.Registry.import_current_user_services()
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+transaction_services = models.Registry.import_transaction_services()
+
+
+def _activity_should_be_deleted(activity_summary, user_id):
```

Comment:
Perhaps move this to the relevant services layer module, and call it is_private_and_solely_owned_by_user(user_id) -- or, better still, make this a method on the activity rights/summary object.